contrib/netbox.service

```diff
@@ -0,0 +1,24 @@
+[Unit]
+Description=Netbox WSGI
+Documentation=https://netbox.readthedocs.io/en/stable/
+After=network-online.target
+Wants=network-online.target
+
+[Service]
+Type=simple
+
+EnvironmentFile=/etc/sysconfig/netbox.env
+
+User=www-data
+Group=www-data
+PIDFile=${PidPath}
```

Comment:
I do not think PIDFile option should be set.  It's only used if the application itself forks and writes out a pidfile itself.  For full details see [this page](https://www.freedesktop.org/software/systemd/man/systemd.service.html)

However, gunicorn does not fork in our current config (daemon=false by default), which is correct.

Reply:
A little incorrect.  PIDFile is used irregardless of whether it is sett to type=fork, if it is included.

Reply:
I would agree that the [documentation](https://www.freedesktop.org/software/systemd/man/systemd.service.html#Options) is very unclear. 

> Usage of this option is recommended for services where `Type=` is set to `forking`.
> ...
> The service manager will read the PID of the main process of the service from this file after start-up of the service

So if you set this option for a non-forking service, it's not recommended (although not forbidden).  Does systemd actually read the PIDFile in this case?  I suppose it *could*, but surely it would ignore the contents - it already knows with certainty the PID of the child, since it spawned it itself.

But I think the strongest argument is: do any other system services use the PIDFile setting when it's not required?  Here is an Ubuntu 18.04 system:

```
root@nuc1:~# ls /lib/systemd/system/*.service | wc -l
210
root@nuc1:~# grep PIDFile /lib/systemd/system/*.service
/lib/systemd/system/apcupsd.service:PIDFile=/var/run/apcupsd.pid
/lib/systemd/system/dm-event.service:PIDFile=/run/dmeventd.pid
/lib/systemd/system/iscsid.service:PIDFile=/run/iscsid.pid
/lib/systemd/system/lvm2-lvmetad.service:PIDFile=/run/lvmetad.pid
/lib/systemd/system/lvm2-lvmpolld.service:PIDFile=/run/lvmpolld.pid
/lib/systemd/system/mdmon@.service:# Don't set the PIDFile.  It isn't necessary (systemd can work
/lib/systemd/system/mdmon@.service:#PIDFile=/run/mdadm/%I.pid
/lib/systemd/system/rpc-statd.service:PIDFile=/var/run/rpc.statd.pid
root@nuc1:~#
```

Out of 210 services, 6 of them use PIDFile - as far as I can tell, these are legacy services where the init script daemonizes the process.  And one script explicitly says don't use PIDFile.

So I believe best practice is - don't use it, unless you have a legacy application which daemonizes itself.